SSSDConfig: Update of config options

alexal opened the pull request:
Added missing config options in SSSDConfig.py along with their translatable descriptions

Resolves: https://pagure.io/SSSD/sssd/issue/1362

alexal commented:
I've found that these options are missing in SSSDConfig.py:

cache_credentials_minimal_first_factor_length
config_file_version
create_homedir
description
entry_cache_nowait_percentage
entry_cache_ssh_host_timeout
get_domains_timeout
groups	
homedir_substring	
homedir_umask
ipa_host_fqdn	
ipa_host_member_of	
ipa_host_name	
ipa_host_object_class	
ipa_host_search_base	
ipa_host_serverhostname	
ipa_host_ssh_public_key	
ipa_host_uuid	
ipa_hostgroup_member	
ipa_hostgroup_memberof	
ipa_hostgroup_name	
ipa_hostgroup_objectclass	
ipa_hostgroup_uuid
ipa_netgroup_domain	
ipa_netgroup_member	
ipa_netgroup_member_ext_host	
ipa_netgroup_member_host	
ipa_netgroup_member_of	
ipa_netgroup_member_user	
ipa_netgroup_name	
ipa_netgroup_object_class	
ipa_netgroup_uuid
ipa_selinux_usermap_enabled	
ipa_selinux_usermap_host_category	
ipa_selinux_usermap_member_host	
ipa_selinux_usermap_member_user	
ipa_selinux_usermap_name	
ipa_selinux_usermap_object_class	
ipa_selinux_usermap_see_also	
ipa_selinux_usermap_selinux_user	
ipa_selinux_usermap_user_category	
ipa_selinux_usermap_uuid
ipa_server_mode	
ipa_subdomains_search_base	
ipa_sudocmd_memberof	
ipa_sudocmd_object_class	
ipa_sudocmd_sudoCmd	
ipa_sudocmd_uuid	
ipa_sudocmdgroup_entry_usn	
ipa_sudocmdgroup_member	
ipa_sudocmdgroup_name	
ipa_sudocmdgroup_object_class	
ipa_sudocmdgroup_uuid	
ipa_sudorule_allowcmd	
ipa_sudorule_cmdcategory	
ipa_sudorule_denycmd	
ipa_sudorule_enabled_flag	
ipa_sudorule_entry_usn	
ipa_sudorule_externaluser	
ipa_sudorule_host	
ipa_sudorule_hostcategory	
ipa_sudorule_name	
ipa_sudorule_notafter	
ipa_sudorule_notbefore	
ipa_sudorule_object_class	
ipa_sudorule_option	
ipa_sudorule_runasextgroup	
ipa_sudorule_runasextuser	
ipa_sudorule_runasextusergroup	
ipa_sudorule_runasgroup	
ipa_sudorule_runasgroupcategory	
ipa_sudorule_runasusercategory	
ipa_sudorule_sudoorder	
ipa_sudorule_user	
ipa_sudorule_usercategory	
ipa_sudorule_uuid
ldap_autofs_map_master_name
ldap_group_entry_usn
ldap_group_search_filter	
ldap_group_search_scope
ldap_rfc2307_fallback_to_local_users
ldap_service_entry_usn
ldap_user_entry_usn
mail_dir
monitor_resolv_conf
offline_timeout
pwd_expiration_warning
realmd_tags
remove_homedir
scope
simple_allow_groups
simple_deny_groups
skel_dir
subdomains_provider
try_inotify
userdel_cmd
users

I've added most of them with the exception of these shown below, for which I wasn't able to find descriptions in any man pages:

ipa_host_member_of
ipa_host_name
ipa_host_serverhostname
ipa_host_uuid
ipa_hostgroup_member
ipa_hostgroup_memberof
ipa_hostgroup_name
ipa_hostgroup_objectclass
ipa_hostgroup_uuid
ipa_sudocmd_memberof	
ipa_sudocmd_object_class	
ipa_sudocmd_sudoCmd	
ipa_sudocmd_uuid	
ipa_sudocmdgroup_entry_usn	
ipa_sudocmdgroup_member
ipa_sudocmdgroup_name	
ipa_sudocmdgroup_object_class	
ipa_sudocmdgroup_uuid	
ipa_sudorule_allowcmd	
ipa_sudorule_cmdcategory	
ipa_sudorule_denycmd	
ipa_sudorule_enabled_flag	
ipa_sudorule_entry_usn	
ipa_sudorule_externaluser	
ipa_sudorule_host	
ipa_sudorule_hostcategory	
ipa_sudorule_name	
ipa_sudorule_notafter	
ipa_sudorule_notbefore	
ipa_sudorule_object_class	
ipa_sudorule_option	
ipa_sudorule_runasextgroup	
ipa_sudorule_runasextuser	
ipa_sudorule_runasextusergroup	
ipa_sudorule_runasgroup	
ipa_sudorule_runasgroupcategory	
ipa_sudorule_runasusercategory	
ipa_sudorule_sudoorder	
ipa_sudorule_user	
ipa_sudorule_usercategory	
ipa_sudorule_uuid
ldap_group_search_filter
ldap_group_entry_usn
ldap_group_search_scope
ldap_service_entry_usn
ldap_user_entry_usn

pbrezina commented:
Thank you.

Remaining `ipa_*` options are just LDAP attributes names. These should not be changed so I think it is fine to leave those without description.

`_entry_usn` are replaced by single `ldap_entry_usn` so no need to describe them here.

The rest can be derived like this:
`ldap_user_search_filter` -> `ldap_group_search_filter`
`ldap_user_search_scope` -. `ldap_group_search_scope`

Please add those two.

pbrezina commented:
Also may I ask you to change the commit title to something that makes clear what is changed? (Not only what file is touched.) Thank you.

alexal commented:
@pbrezina thanks for your comments. I've made changes and updated the commit message to:

 - Added missing config options with a description
 - Removed not used or replaced options such as ldap_group_search_scope, ldap_group_search_filter, etc...

pbrezina commented:
Thank you. Ack.

pbrezina commented:
* `master`
  * 9483bf410eeb5de2fee3a4a61a80d96682772d51 - SSSDConfig: Update of config options
